File: SpringBootApp/src/main/java/com/example/springbootapp/data/dto/AddToCartRequestDto.java
package com.example.springbootapp.data.dto;

import com.example.springbootapp.data.entities.Embdeddables.Personalization;
import com.example.springbootapp.data.entities.Enums.ProductCategory;
import com.example.springbootapp.data.entities.Enums.Size;
import com.example.springbootapp.security.customValidators.ValidPersonalization;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@ValidPersonalization   // Annotazione custom
public class AddToCartRequestDto {
    @NotBlank
    private String productId;
    @NotNull
    private Size size;
    @NotNull
    private ProductCategory category;
    private Personalization personalization;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/dto/AddToCartRequestDto.kt
package com.example.clientuser.model.dto

import com.example.clientuser.model.Personalization
import com.example.clientuser.model.enumerator.ProductCategory
import com.example.clientuser.model.enumerator.Size
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class AddToCartRequestDto(
    val productId: String,
    val size: Size,
    val category: ProductCategory,
    val personalization: Personalization
)